lib/src/utils/confirmations/archethic_transaction_sender.dart

```diff
@@ -25,56 +30,123 @@ class ArchethicTransactionSender
   StreamSubscription? _transactionConfirmedSubscription;
   StreamSubscription? _transactionErrorSubscription;
   Timer? _timer;
+  Completer<TransactionConfirmation?>? _completer;
 
   @override
   void close() {
-    _timer?.cancel();
-    _phoenixLink?.dispose();
-    _phoenixHttpLink?.dispose();
-    _transactionConfirmedSubscription?.cancel();
-    _transactionErrorSubscription?.cancel();
+    if (_completer?.isCompleted == false) {
+      _completer!.complete(null);
+    }
+    _reset();
   }
 
-  /// Sends a transaction and listens to confirmations.
-  ///
-  /// Sender auto-closes in the following situations :
-  ///     - when transaction is fully confirmed
-  ///     - when timeout is reached
-  ///     - when transaction fails
   @override
-  Future<void> send({
+  Future<TransactionConfirmation?> send({
```

Comment:
This simplification makes `sendTransactions` pretty much useless as it can be replaced by : 

```dart
for (final tx in transactions) {
  await TransactionSender(apiService).send(tx);
}
```

What do you think about that ?

Reply:
Another thought : sending multiple transactions can be different ways.

 - It can be sequential (like in `transaction_utils.dart`
 - It can be "parallel"
 
 Each case has its own choices to make regarding failure management : should we continue if a Tx send fails ?
 
 I'm not sure that choice should be made by `TransactionSender` for now...